packages/salesforcedx-vscode-apex/test/jest/index.test.ts

```diff
@@ -0,0 +1,37 @@
+import { API } from '../../src/constants';
+import * as index from '../../src/index';
+import { languageClientUtils } from '../../src/languageUtils';
+import { extensionUtils } from '../../src/languageUtils/extensionUtils';
+
+describe('indexDoneHandler', () => {
+  let setStatusSpy: jest.SpyInstance;
+  let onNotificationSpy: jest.SpyInstance;
+  let mockLanguageClient: any;
+  let setClientReadySpy: jest.SpyInstance;
+
+  beforeEach( () => {
+    setStatusSpy = jest.spyOn(languageClientUtils, 'setStatus').mockImplementation(jest.fn());
```

Comment:
I don't think you need to use `mockImplementation` here.  It can just be `jest.spyOn(languageClientUtils, 'setStatus').mockReturnValue(undefined);` This is true for all .mockImplementation calls in this test. 

mockImplementation should only be used when you need to execute some code when a mock'd function is called.

Reply:
One important note.  If you don't set `mockReturnValue` or for async `mockResolvedValue` for mocks it will call through to the original function that is mocked.